### PR TITLE
Importing of configuration fix, updates readme

### PR DESCRIPTION
**Pull request by pdyba**

### CODE REVIEW

File: lbz/_cfg.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from lbz.configuration import EnvValue
+
+# LBZ configuration
+LBZ_DEBUG_MODE = EnvValue("LBZ_DEBUG_MODE", default=False, parser=lambda a: a == "true")
+LOGGING_LEVEL = EnvValue("LOGGING_LEVEL", default="INFO")
+CORS_HEADERS = EnvValue("CORS_HEADERS", default=[], parser=lambda a: a.split(","))
+CORS_ORIGIN = EnvValue("CORS_ORIGIN", default=[], parser=lambda a: a.split(","))
+
+# AWS related configuration
+AWS_LAMBDA_FUNCTION_NAME = EnvValue("AWS_LAMBDA_FUNCTION_NAME", default="lbz-event-api").value
```
Comment (redlickigrzegorz) [inline lbz/_cfg.py:12]: Once this variable became more generic, I don't think we should put the default value to `lbz-event-api`.

Maybe, we shouldn't use the default value at all - this would make it required to be set if this part of the code is used 🤔

File: lbz/_cfg.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+EVENTS_BUS_NAME = EnvValue(
+    "EVENTS_BUS_NAME", default=f"{AWS_LAMBDA_FUNCTION_NAME}-event-bus"
```
Comment (redlickigrzegorz) [inline lbz/_cfg.py:14]: The same here - I don't think the default value is good here.

File: lbz/_cfg.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+).value
```
Comment (redlickigrzegorz) [inline lbz/_cfg.py:15]: Also, resigning from lazy-loading may produce unexpected problems in tests (depending on when exactly these values are set) 🤔

File: lbz/_cfg.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+# Authorization configuration
+PUBLIC_KEYS = EnvValue("ALLOWED_PUBLIC_KEYS", parser=lambda a: json.loads(a)["keys"])
+ALLOWED_AUDIENCES = EnvValue("ALLOWED_AUDIENCES", parser=lambda a: a.split(","))
+ALLOWED_ISS = EnvValue("ALLOWED_ISS", default="")
+REMOVE_PREFIXES = EnvValue("AUTH_REMOVE_PREFIXES", default=False, parser=bool)
```
Comment (redlickigrzegorz) [inline lbz/_cfg.py:21]: I'm not sure if to use `bool` as a parser - it means that only an empty string or an unset value will give us `False`.

On the other hand e.g. `0` and` false` will stand for `True` 🤔

Comment (redlickigrzegorz) [inline lbz/_cfg.py:21]: Also, shouldn't it be named `AUTH_REMOVE_PREFIXES` - the same as the env value?

File: lbz/authentication.py [hunk 5]
```diff
@@ -1,14 +1,11 @@
 """
 JWT based Authentication module.
 """
-import os
-
+from lbz._cfg import REMOVE_PREFIXES
```
Comment (redlickigrzegorz) [inline lbz/authentication.py:4]: Previously, it was a boolean value but now here, you get the `EnvValue` object - you need to use its `.value` to work with it correctly below.

File: lbz/dev/server.py [hunk 6]
```diff
@@ -6,16 +6,12 @@
...
 from http.server import BaseHTTPRequestHandler, ThreadingHTTPServer
-from os import environ
 from threading import Thread
 from typing import Tuple, Type, Union
 
 from lbz.dev.misc import APIGatewayEvent
 from lbz.resource import Resource
 
-if environ.get("LBZ_DEBUG_MODE") is None:
-    environ["LBZ_DEBUG_MODE"] = "true"
```
Comment (redlickigrzegorz) [inline lbz/dev/server.py:17]: It makes a little sense to have it in the dev server 🤔 

At least I remember you added it to always enable debug mode here.

File: lbz/misc.py [hunk 7]
```diff
@@ -131,7 +130,7 @@ def deep_update(dict_to_update: dict, update_data: dict) -> None:
 
 
 def is_in_debug_mode() -> bool:
-    return environ.get("LBZ_DEBUG_MODE", "").lower() == "true"
+    return cast(bool, LBZ_DEBUG_MODE.value)
```
Comment (redlickigrzegorz) [inline lbz/misc.py:133]: If we need to cast the type of `.value`, this means our `ConfigValue` is wrongly implemented 😞

Comment (redlickigrzegorz) [reply lbz/misc.py:133]: Or the parser is not specific enough 🤔

File: lbz/_cfg.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+CORS_ORIGIN = EnvValue("CORS_ORIGIN", default=[], parser=lambda a: a.split(","))
+
+# AWS related configuration
+AWS_LAMBDA_FUNCTION_NAME = EnvValue("AWS_LAMBDA_FUNCTION_NAME", default="lbz-event-api").value
+EVENTS_BUS_NAME = EnvValue(
+    "EVENTS_BUS_NAME", default=f"{AWS_LAMBDA_FUNCTION_NAME}-event-bus"
+).value
+
+# Authorization configuration
+PUBLIC_KEYS = EnvValue("ALLOWED_PUBLIC_KEYS", parser=lambda a: json.loads(a)["keys"])
```
Comment (redlickigrzegorz) [inline lbz/_cfg.py:18]: The same here, shouldn't it be named `ALLOWED_PUBLIC_KEYS`?

File: lbz/resource.py [hunk 9]
```diff
@@ -89,7 +90,11 @@ def __repr__(self) -> str:
         return f"<Resource {self.method} @ {self.urn} >"
 
     def _load_configuration(self) -> None:
-        self.auth_enabled = env.get("ALLOWED_PUBLIC_KEYS") or env.get("ALLOWED_AUDIENCES")
```
Comment (redlickigrzegorz) [inline lbz/resource.py:92]: Previously, this variable was set based on both `ALLOWED_PUBLIC_KEYS` and `ALLOWED_AUDIENCES`.

File: lbz/resource.py [hunk 10]
```diff
@@ -89,7 +90,11 @@ def __repr__(self) -> str:
...
+        try:
+            PUBLIC_KEYS.value  # pylint: disable=pointless-statement
+            self.auth_enabled = True
+        except MissingConfigValue:
```
Comment (redlickigrzegorz) [inline lbz/resource.py:96]: We should never base on this Exception, so, is `ALLOWED_PUBLIC_KEYS` required or not?

File: lbz/resource.py [hunk 11]
```diff
@@ -154,13 +159,10 @@ def __init__(
     ):
         # TODO: adjust the rest of the arguments in the near future too.
         super().__init__(event)
-        cors_headers = cors_headers or []
-        if not cors_headers and (env_headers := env.get("CORS_HEADERS")):
-            cors_headers = env_headers.split(",")
+        if not cors_headers:
+            cors_headers = cast(list, CORS_HEADERS.value)
```
Comment (redlickigrzegorz) [inline lbz/resource.py:163]: Here we have the same problem - it is very possible that we should not use lambda functions as parsers 🤔

File: lbz/resource.py [hunk 12]
```diff
@@ -154,13 +159,10 @@ def __init__(
     ):
         # TODO: adjust the rest of the arguments in the near future too.
         super().__init__(event)
-        cors_headers = cors_headers or []
```
Comment (redlickigrzegorz) [inline lbz/resource.py:157]: This line was pretty important as the code below assumes that `cors_headers` is always a list.

File: tests/test_authentication.py [hunk 13]
```diff
@@ -29,14 +30,16 @@ def test_decoding_user_raises_unauthorized_when_invalid_token(user_token: str) -
 
 
 def test_decoding_user_raises_unauthorized_when_invalid_audience(user_token: str) -> None:
-    with pytest.raises(Unauthorized), patch("lbz.jwt_utils.ALLOWED_AUDIENCES", [str(uuid4())]):
+    with pytest.raises(Unauthorized), patch(
+        "lbz.jwt_utils.ALLOWED_AUDIENCES", MockedConfig(str(uuid4()), parser=lambda a: [a])
```
Comment (redlickigrzegorz) [inline tests/test_authentication.py:34]: It was done this way before as this variable was declared globally. Now, when we base it on configuration values, it shouldn't be needed.

But to remove all these mocks or to make them better, I would need to investigate what they really want to replace 🤔

File: tests/test_jwt_utils.py [hunk 14]
```diff
@@ -63,12 +63,6 @@ def test_expired_jwt(self) -> None:
         with pytest.raises(Unauthorized, match="Your token has expired. Please refresh it."):
             decode_jwt(jwt_token)
 
-    @patch("lbz.jwt_utils.get_matching_jwk", MagicMock(return_value={}))
-    @patch("lbz.jwt_utils.ALLOWED_AUDIENCES", [])
-    def test_empty_allowed_audiences(self) -> None:
-        with pytest.raises(RuntimeError, match="ALLOWED_AUDIENCES"):
```
Comment (redlickigrzegorz) [inline tests/test_jwt_utils.py:69]: In general, the logic remains but we raise a different Exception 🤔 

I would leave these tests and only adjust them, otherwise, we can easily misconfigure lbz by setting a default value for these two variables.

File: tests/test_resource.py [hunk 15]
```diff
@@ -370,7 +371,10 @@ def test_cors_headers_param_request(self) -> None:
             ALLOW_ORIGIN_HEADER: ORIGIN_EXAMPLE,
         }
 
-    @patch.dict(environ, {"CORS_HEADERS": "X-PRN-KEY,X-PRN-TOKEN"})
+    @patch(
+        "lbz.resource.CORS_HEADERS",
+        MockedConfig("x", parser=lambda _a: ["X-PRN-KEY", "X-PRN-TOKEN"]),
```
Comment (redlickigrzegorz) [inline tests/test_resource.py:376]: I have a feeling that we need to add the possibility to clear configuration 🤔

Each test should be treated more or less separately, otherwise, workarounds like this one will be added randomly in the code - unfortunately, this is the wrong approach.

File: tests/conftest.py [hunk 16]
```diff
@@ -20,6 +21,11 @@
 from tests.utils import encode_token
 
 
+class MockedConfig(ConfigValue):
```
Comment (redlickigrzegorz) [inline tests/conftest.py:24]: Based on the above comments, this class should be removed.

**Review by redlickigrzegorz — CHANGES_REQUESTED** [review]

File: lbz/resource.py (hunk 9 above)
Comment (pdyba) [reply lbz/resource.py:92]: I know but it was wrong.

File: lbz/_cfg.py (hunk 2 above)
Comment (pdyba) [reply lbz/_cfg.py:14]: Agreed remove default

File: lbz/resource.py (hunk 12 above)
Comment (pdyba) [reply lbz/resource.py:157]: and it is - default ensures that.

File: lbz/configuration.py [hunk 17]
```diff
@@ -1,13 +1,30 @@
...
 T = TypeVar("T")
 
 
+class ConfigParser:
+    @staticmethod
+    def split_by_comma(cfg: str) -> List[str]:
+        return cfg.split(",")
+
+    @staticmethod
+    def env_bool(cfg: str) -> bool:
```
Comment (redlickigrzegorz) [inline lbz/configuration.py:18]: Why `env`? Exactly the same string could be stored in SSM 😉

Comment (redlickigrzegorz) [reply lbz/configuration.py:18]: Also, to keep naming consistent, it should be rather named as a command e.g. `cast_to_bool` or something similar.

File: lbz/configuration.py [hunk 18]
```diff
@@ -1,13 +1,30 @@
...
+        if cfg.lower() in ("true", "1"):
+            return True
+        return False
+
+    @staticmethod
+    def load_jwt_keys(cfg: str) -> List[dict]:
+        return cast(list, json.loads(cfg)["keys"])
```
Comment (redlickigrzegorz) [inline lbz/configuration.py:25]: Using `cast` is not a silver bullet. First you should think why `mypy` cannot recognize the type and try to help him 😉 

e.g.
```python
deserialized_value: Dict[str, List[dict]] = json.loads(value)
return deserialized_value["keys"]
```

You can even create a custom `TypedDict` instead of declaring it this way - it might be useful in other places.

File: lbz/configuration.py [hunk 19]
```diff
@@ -1,13 +1,30 @@
...
+        return cfg.split(",")
+
+    @staticmethod
+    def env_bool(cfg: str) -> bool:
+        if cfg.lower() in ("true", "1"):
+            return True
+        return False
+
+    @staticmethod
+    def load_jwt_keys(cfg: str) -> List[dict]:
```
Comment (redlickigrzegorz) [inline lbz/configuration.py:24]: Also, it would be good to write simple tests for these parsers - it will simplify checking what is expected as an input/output 😉

File: lbz/configuration.py [hunk 20]
```diff
@@ -47,6 +64,9 @@ def _parse_value(self, value: Any) -> T:
         except Exception as error:
             raise ConfigValueParsingFailed(self._key, value) from error
 
+    def reset(self) -> None:
```
Comment (redlickigrzegorz) [inline lbz/configuration.py:67]: Also, I would write one test for this newly added method - we should avoid adding public methods that are not tested, even if they are simple.

File: tests/conftest.py [hunk 21]
```diff
@@ -16,13 +29,41 @@
...
+            "AUTH_REMOVE_PREFIXES": "1",
+            "ALLOWED_PUBLIC_KEYS": json.dumps({"keys": [SAMPLE_PUBLIC_KEY]}),
+            "ALLOWED_AUDIENCES": ",".join(allowed_audiences),
+            "ALLOWED_ISS": "test-issuer",
+            "AWS_LAMBDA_FUNCTION_NAME": "million-dollar-lambda",
+            "EVENTS_BUS_NAME": "million-dollar-lambda-event-bus",
+            "AWS_DEFAULT_REGION": "us-west-2",
+        },
+        clear=True,
+    ):
```
Comment (redlickigrzegorz) [inline tests/conftest.py:55]: As I mentioned in the past, avoid creating multiline context managers - you can always declare a variable above 😉

File: tests/conftest.py [hunk 22]
```diff
@@ -16,13 +29,41 @@
 from lbz.resource import Resource
 from lbz.response import Response
 from lbz.router import Router, add_route
-from tests import SAMPLE_PRIVATE_KEY
+from tests.fixtures.rsa_pair import SAMPLE_PRIVATE_KEY, SAMPLE_PUBLIC_KEY
```
Comment (redlickigrzegorz) [inline tests/conftest.py:32]: If you imported that directly from the "_fixtures_", do we need to import them additionally in `tests/__init__.py`?

File: tests/conftest.py [hunk 23]
```diff
@@ -16,13 +29,41 @@
...
 
 
 @pytest.fixture(scope="session", name="allowed_audiences")
 def allowed_audiences_fixture() -> List[str]:
-    return os.environ["ALLOWED_AUDIENCES"].split(",")
+    return [str(uuid4()), str(uuid4())]
+
+
+@pytest.fixture(autouse=True, name="patch_environ")
+def patch_environ_fixture(allowed_audiences: List[str]) -> Iterator[None]:
```
Comment (redlickigrzegorz) [inline tests/conftest.py:42]: `environ` is patched, that's true, but I would name this fixture according to what is expected, e.g.:
```python
def setting_initial_lbz_configuration(...) -> Iterator[None]:
```

File: tests/conftest.py [hunk 24]
```diff
@@ -16,13 +29,41 @@
...
 from tests.utils import encode_token
 
 
 @pytest.fixture(scope="session", name="allowed_audiences")
 def allowed_audiences_fixture() -> List[str]:
-    return os.environ["ALLOWED_AUDIENCES"].split(",")
+    return [str(uuid4()), str(uuid4())]
+
+
+@pytest.fixture(autouse=True, name="patch_environ")
```
Comment (redlickigrzegorz) [inline tests/conftest.py:41]: You don't have to declare `name` if you don't want to use it below 😉

File: tests/conftest.py [hunk 25]
```diff
@@ -133,7 +174,7 @@ def user_token_fixture(user_cognito: dict) -> str:
     return encode_token(user_cognito)
 
 
-@pytest.fixture(scope="session", name="user")
+@pytest.fixture(name="user")  # scope="session", - TODO: bring that back to reduce run time
```
Comment (redlickigrzegorz) [inline tests/conftest.py:177]: Will we be able to bring that back if the `User` depends on configuration? I think a different local fixture will be required anyway to satisfy the problematic tests.

File: tests/test_jwt_utils.py [hunk 26]
```diff
@@ -78,15 +74,22 @@ def test_missing_correct_audiences(self, caplog: pytest.LogCaptureFixture) -> No
...
-    def test_empty_public_keys(self) -> None:
-        with pytest.raises(RuntimeError):
-            decode_jwt("x")
-
     def test_validate_missing_iss_exception(self) -> None:
         with pytest.raises(SecurityError, match="'exp'"):
             validate_jwt_properties({"allow": "*", "deny": {}})
 
+    @patch.dict(environ, {}, clear=True)
+    def test_empty_public_keys(self) -> None:
```
Comment (redlickigrzegorz) [inline tests/test_jwt_utils.py:82]: Just out of curiosity, why did you move these two tests?

Comment (pdyba) [reply tests/test_jwt_utils.py:82]: I think they were even removed for a second and then restored - no actual reason at the end.

File: lbz/jwt_utils.py [hunk 27]
```diff
@@ -1,40 +1,27 @@
...
     """
     Checks provided JWT token against allowed tokens.
     """
     try:
         kid_from_jwt_header = jwt.get_unverified_header(auth_jwt_token)["kid"]
-        for key in PUBLIC_KEYS:
+        for key in ALLOWED_PUBLIC_KEYS.value:
             if key["kid"] == kid_from_jwt_header:
-                return key
+                return cast(dict, key)
```
Comment (redlickigrzegorz) [inline lbz/jwt_utils.py:24]: When you improve typing, you won't have to cast this variable.

File: lbz/jwt_utils.py [hunk 28]
```diff
@@ -54,29 +41,32 @@ def validate_jwt_properties(decoded_jwt: dict) -> None:
...
 
-    if not ALLOWED_AUDIENCES:
-        raise RuntimeError("ALLOWED_AUDIENCES were not defined")
+    if not ALLOWED_PUBLIC_KEYS.value:
+        raise MissingConfigValue("ALLOWED_PUBLIC_KEYS")
+
+    if not ALLOWED_AUDIENCES.value:
+        raise MissingConfigValue("ALLOWED_AUDIENCES")
+
+    if any("kid" not in public_key for public_key in ALLOWED_PUBLIC_KEYS.value):
```
Comment (redlickigrzegorz) [inline lbz/jwt_utils.py:59]: This line checks the correctness of the environment variable, so, I think it should be implemented directly in the parser: `ConfigParser.load_jwt_keys()` 😉

File: tests/test_jwt_utils.py [hunk 29]
```diff
@@ -78,15 +74,22 @@ def test_missing_correct_audiences(self, caplog: pytest.LogCaptureFixture) -> No
...
+        with pytest.raises(MissingConfigValue, match="ALLOWED_PUBLIC_KEYS"):
```
Comment (redlickigrzegorz) [inline tests/test_jwt_utils.py:83]: This is the message you expect to see:
```suggestion
        with pytest.raises(MissingConfigValue, match="'ALLOWED_PUBLIC_KEYS' was not defined."):
```

File: tests/test_jwt_utils.py [hunk 30]
```diff
@@ -78,15 +74,22 @@ def test_missing_correct_audiences(self, caplog: pytest.LogCaptureFixture) -> No
...
+            decode_jwt("x")
+
+    @patch.dict(
+        environ, {"ALLOWED_PUBLIC_KEYS": json.dumps({"keys": [SAMPLE_PUBLIC_KEY]})}, clear=True
+    )
+    def test_empty_allowed_audiences(self) -> None:
+        with pytest.raises(MissingConfigValue, match="ALLOWED_AUDIENCES"):
```
Comment (redlickigrzegorz) [inline tests/test_jwt_utils.py:90]: The same here:
```suggestion
        with pytest.raises(MissingConfigValue, match="'ALLOWED_AUDIENCES' was not defined."):
```

File: tests/test_configuration.py [hunk 31]
```diff
@@ -67,11 +70,22 @@ def test_gets_value_from_env_with_json_parser(self) -> None:
...
 
+    @patch.dict(environ, {"RANDOM_VAR": "not-none"})
+    def test_reset_sets_value_to_none_forcing_fetching_it_once_again(self) -> None:
+        env_cfg = EnvValue[str]("RANDOM_VAR")
+        assert env_cfg.value == "not-none"
+
+        env_cfg.reset()
+
+        with patch.dict(environ, {"RANDOM_VAR": "still-not-none"}):
+            env_cfg = EnvValue[str]("RANDOM_VAR")
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:81]: You cannot create a completely new variable as this won't confirm the `.reset()` method logic.

File: tests/test_configuration.py [hunk 32]
```diff
@@ -29,14 +29,17 @@ def test_returns_default_when_no_value_set(self) -> None:
...
-        cfg = EnvValue("test")
+        cfg = EnvValue[str]("test")
 
         with pytest.raises(MissingConfigValue, match="'test' was not defined."):
             cfg.value  # pylint: disable=pointless-statement
 
     @patch.object(EnvValue, "getter", MagicMock(return_value=1))
     def test_modifies_config_value_using_declared_parser_function(self) -> None:
-        cfg = EnvValue("RANDOM_VAR", lambda a: a * 10)
+        def times_10_parser(value: int) -> int:
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:39]: `value` should be always a string, as it is impossible to get something else from env:
```suggestion
        def times_10_parser(value: str) -> int:
```

File: tests/test_configuration.py [hunk 33]
```diff
@@ -29,14 +29,17 @@ def test_returns_default_when_no_value_set(self) -> None:
         assert cfg.value == 42
 
     def test_raises_missing_configuration_when_value_is_none(self) -> None:
-        cfg = EnvValue("test")
+        cfg = EnvValue[str]("test")
 
         with pytest.raises(MissingConfigValue, match="'test' was not defined."):
             cfg.value  # pylint: disable=pointless-statement
 
     @patch.object(EnvValue, "getter", MagicMock(return_value=1))
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:37]: To make it more correct you should patch environ instead:
```suggestion
    @patch.dict(environ, {"RANDOM_VAR": "1"})
```

File: tests/test_configuration.py [hunk 34]
```diff
@@ -29,14 +29,17 @@ def test_returns_default_when_no_value_set(self) -> None:
...
     def test_modifies_config_value_using_declared_parser_function(self) -> None:
-        cfg = EnvValue("RANDOM_VAR", lambda a: a * 10)
+        def times_10_parser(value: int) -> int:
+            return value * 10
+
+        cfg = EnvValue[int]("RANDOM_VAR", times_10_parser)
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:42]:
```suggestion
        cfg = EnvValue("RANDOM_VAR", times_10_parser)
```

File: tests/test_configuration.py [hunk 35]
```diff
@@ -80,8 +94,34 @@ class TestSSMConfig:
...
-        cfg = SSMValue("key_name")
+        cfg = SSMValue[str]("key_name")
 
         assert cfg.value == "test_value"
 
         mocked_get_parameter.assert_called_once_with("key_name")
+
+
+class TestConfigParser:
+    def test_split_by_comma_returns_list(self) -> None:
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:105]:
```suggestion
    def test__split_by_comma__returns_list(self) -> None:
```

File: tests/test_configuration.py [hunk 36]
```diff
@@ -80,8 +94,34 @@ class TestSSMConfig:
...
+            ("TruE", True),
+            ("0", False),
+            ("False", False),
+            ("fLase", False),
+            ("not-true", False),
+            ("N/A", False),
+            ("xxxxx", False),
+        ],
+    )
+    def test_cast_to_bool_returns_bool(self, input_value: str, expected_value: bool) -> None:
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:123]:
```suggestion
    def test__cast_to_bool__returns_bool(self, input_value: str, expected_value: bool) -> None:
```

File: tests/test_configuration.py [hunk 37]
```diff
@@ -80,8 +94,34 @@ class TestSSMConfig:
...
+        assert ConfigParser.cast_to_bool(input_value) == expected_value
+
+    def test_load_jwt_keys_return_value_of_keys(self) -> None:
```
Comment (redlickigrzegorz) [inline tests/test_configuration.py:126]:
```suggestion
    def test__load_jwt_keys__return_value_of_keys(self) -> None:
```

**Review by redlickigrzegorz — APPROVED** [review]

File: lbz/_cfg.py [hunk 38]
```diff
@@ -0,0 +1,23 @@
...
+
+# LBZ configuration
+LBZ_DEBUG_MODE = EnvValue[bool]("LBZ_DEBUG_MODE", default=False, parser=ConfigParser.cast_to_bool)
+LOGGING_LEVEL = EnvValue("LOGGING_LEVEL", default="INFO")
+CORS_HEADERS = EnvValue[List[str]]("CORS_HEADERS", default=[], parser=ConfigParser.split_by_comma)
+CORS_ORIGIN = EnvValue[List[str]]("CORS_ORIGIN", default=[], parser=ConfigParser.split_by_comma)
+
+# AWS related configuration
+AWS_LAMBDA_FUNCTION_NAME = EnvValue[str]("AWS_LAMBDA_FUNCTION_NAME")
+EVENTS_BUS_NAME = EnvValue[str]("EVENTS_BUS_NAME")
```
[the remark this answers is not in the document]
Comment (pdyba) [reply lbz/_cfg.py:13]: Actually all of the EnvValue started to expect that type definition after recent change - mypy is not happy if the don't have it.

[the remark this answers is not in the document]
Comment (pdyba) [reply lbz/_cfg.py:13]: When there is no default value or parser defined ;/
